### PR TITLE
Move code from sdformat to gz-usd

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

## Summary

Move code from sdformat https://github.com/gazebosim/sdformat/pull/1094 to gz-usd.

Mainly adapted `CMakeLists.txt` and namespaces

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by scpeters — COMMENTED** [review]
This is currently using fortress dependencies, but I expect that we may have a `garden` branch at some point in the future with the newer dependencies.

Thinking about garden, I'd rather see `gz` namespaces instead of `ignition` to more easily maintain forward compatibility (sorry!).

**Comment by ahcorde** [comment]
> This is currently using fortress dependencies, but I expect that we may have a `garden` branch at some point in the future with the newer dependencies.
> 
> Thinking about garden, I'd rather see `gz` namespaces instead of `ignition` to more easily maintain forward compatibility (sorry!).

The idea that I have in mind it's to create `gz-usd2` branch for `garden` as a follow up PR

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.10.2 FATAL_ERROR)
+
+#============================================================================
+# Initialize the project
+#============================================================================
+project(ignition-usd1 VERSION 1.0.0)
```
Comment (chapulina) [inline CMakeLists.txt:6]: You could keep it in major version zero until it's stable. We've done this with core Gazebo libraries in the past, Acropolis shipped with `ign-launch0`, so we could break API / ABI within Acropolis

Comment (ahcorde) [reply CMakeLists.txt]: https://github.com/gazebosim/gz-usd/pull/1/commits/a21d5b60525204df1c8157491b0578ac41289c7c

**Review by scpeters — APPROVED** [review]

**Pull request merged**